src/arrayMethodSplice.js

```diff
@@ -5,7 +5,54 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    let searchFrom;
+    let copiedItemsFromStartIndex = [];
+    const deletedItems = [];
+    let deleteLength;
+
+    if ((start < 0 && (this.length + start) < 0)
+    || (!start && arguments.length)) {
+      searchFrom = 0;
+    } else if (start < 0) {
+      searchFrom = this.length + start;
+    } else if (start > this.length) {
+      searchFrom = this.length;
+    } else {
+      searchFrom = start;
+    }
+
+    for (let i = searchFrom; i < this.length; i++) {
+      copiedItemsFromStartIndex = [ ...copiedItemsFromStartIndex, this[i] ];
```

Comment:
just noticed you don't need to copy the whole array, therefore you don't need spread operator here and you can go back to your previous solution on this line